Gems/ROS2/Code/Source/Camera/CameraPublishers.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "CameraConstants.h"
+#include "CameraPublishers.h"
+#include "CameraSensor.h"
+#include <ROS2/Communication/TopicConfiguration.h>
+#include <ROS2/Utilities/ROS2Names.h>
+#include <ROS2/ROS2Bus.h>
+
+namespace ROS2
+{
+    namespace Internal
+    {
+        using TopicConfigurations = AZStd::unordered_map<CameraSensorDescription::CameraChannelType, TopicConfiguration>;
+
+        template<typename CameraType>
+        TopicConfigurations GetCameraTopicConfiguration([[maybe_unused]] const SensorConfiguration& sensorConfiguration)
+        {
+            AZ_Error("GetCameraTopicConfiguration", false, "Invalid camera template type!");
+            TopicConfigurations tc;
+            return tc;
```

Comment:
minor suggestion: You don't need to declare a variable
```suggestion
            return TopicConfigurations();
```